src/Npgsql/ConnectorPool.cs

```diff
@@ -257,8 +258,8 @@ internal async ValueTask<NpgsqlConnector> AllocateLong(NpgsqlConnection conn, Np
                     }
                     Debug.Assert(connector.PoolIndex != int.MaxValue);
 
-                    // Only when we are the ones that incremented openCount past _min Change the timer.
-                    if (openCount == _min)
+                    // Only start pruning if it was this thread that incremented open count past _min.
+                    if (prevOpenCount == _min)
```

Comment:
We don't seem to be doing anything with `openCount` anymore, after assigning it to `prevOpenCount` above. We could maybe simplify by renaming `openCount` to `prevOpenCount` and just operating directly on that the whole way? Less variables, less complexity...

Reply:
It's not really 'the current' open count that you get back from CompareExchange though...

Reply:
Left it for now